apps/query-server/queryserver/main.py

```diff
@@ -67,29 +97,157 @@ async def get_index(
 ) -> Index:
     """Return details on the given index."""
 
-    schema = util.get_schema(sqclient, index)
+    schema = sqclient.get_opensearch_schema(index)
     return Index(index=index, index_schema=schema)
 
 
 @app.post("/v1/plan")
 async def generate_plan(query: Query) -> LogicalPlan:
     """Generate a query plan for the given query, but do not run it."""
 
-    plan = sqclient.generate_plan(query.query, query.index, util.get_schema(sqclient, query.index))
+    if query.query is None:
+        raise ValueError("query is required")
+    if query.plan is not None:
+        raise ValueError("plan must not be specified")
```

Comment:
Why is this necessary?  You don't do this check on query.plan in `run_query_stream`.

Reply:
generate_plan does not allow the plan to be specified up front. But run_query and run_query_stream do allow a pre-defined plan. Make sense?